plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -212,6 +217,111 @@ public void testDescribeTable()
         assertEquals(actualColumns, expectedColumns);
     }
 
+    @Test
+    public void testMaterializedViewsMetadata()
+    {
+        computeActual(copyTableSql("tpch.tiny.lineitem", "iceberg.tpch.lineitem"));
+        computeActual(copyTableSql("tpch.tiny.orders", "iceberg.tpch.orders"));
+
+        computeActual(createMaterializedViewSql());
+        MaterializedResult showStaleMaterializedViewsResult = computeActual(showMaterializedViewsSql());
+
+        assertThat(showStaleMaterializedViewsResult.getRowCount())
+                .isEqualTo(1);
+
+        assertThat(showStaleMaterializedViewsResult.getTypes())
+                .isEqualTo(ImmutableList.of(VARCHAR, VARCHAR, VARCHAR, VARCHAR, VARCHAR, VARCHAR, BOOLEAN, VARCHAR, VARCHAR, VARCHAR));
+
+        assertMaterializedViewRowEquals(
+                showStaleMaterializedViewsResult.getMaterializedRows().get(0),
+                getMaterializedViewsResultRow(false));
+
+        computeActual(format("REFRESH MATERIALIZED VIEW %s", MATERIALIZED_VIEW_NAME));
+        MaterializedResult showFreshMaterializedViewsResult = computeActual(showMaterializedViewsSql());
+
+        assertThat(showFreshMaterializedViewsResult.getRowCount())
+                .isEqualTo(1);
+
+        assertMaterializedViewRowEquals(
+                showFreshMaterializedViewsResult.getMaterializedRows().get(0),
+                getMaterializedViewsResultRow(true));
+
+        computeActual(format("DROP MATERIALIZED VIEW %s", MATERIALIZED_VIEW_NAME));
+
+        assertThat(query(showMaterializedViewsSql()))
+                .returnsEmptyResult();
+    }
+
+    private static void assertMaterializedViewRowEquals(MaterializedRow actual, MaterializedRow expected)
```

Comment:
This should not be needed, we should stick to the usual assertions for comparing rows